debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/jdbc/MySqlConnection.java

```diff
@@ -50,7 +53,13 @@ public boolean isGtidModeEnabled() {
     @Override
     public GtidSet knownGtidSet() {
         try {
-            return queryAndMap("SHOW MASTER STATUS", rs -> {
+            String showMasterStmt = "SHOW MASTER STATUS";
+
+            if (complianceMode.equals(MySqlConnectorConfig.ComplianceMode.MYSQL_84_PLUS)) {
+                showMasterStmt = "SHOW BINARY LOG STATUS;";
+            }
```

Comment:
We don't need to necessarily do it now, but I wonder if we're going to see this type of behavior in the future with various MySQL versions if we should consider `ComplianceMode` returning an object that we can call somewhat like we do with the `StreamingAdapter` in the Oracle connector to abstract away these details of the if/else.  This way in the call sites it's simply `getComplianceMode().getShowMasterStatement()` or something like this.  This way it avoids the redundancy of this logic at each call site.  wdyt?

Reply:
I think that is a great idea. TBH, this was just an initial implementation to figure out what changes I needed to make in our tests.